crates/stages/src/stages/execution.rs

```diff
@@ -403,7 +405,7 @@ mod tests {
         let state_db = create_test_db::<WriteMap>(EnvKind::RW);
         let mut tx = Transaction::new(state_db.as_ref()).unwrap();
         let input = ExecInput {
-            previous_stage: None,
+            previous_stage: Some((PREV_STAGE_ID, 1)),
```

Comment:
NOTE: The `Execution` stage was previously relying solemnly on `BATCH_SIZE` and db state completely ignoring the `previous_stage` parameter of `ExecInput`.

Reply:
The bad assumption that the canonical header table would only contain max number of executed headers. Good catch!